rs/tests/consensus/consensus_performance.rs

```diff
@@ -377,6 +391,64 @@ async fn get_consensus_metrics(nodes: &[IcNodeSnapshot]) -> ConsensusMetrics {
     }
 }
 
+async fn persist_metrics(
+    ic_version: String,
+    metrics: TestMetrics,
+    message_size: usize,
+    rps: f64,
+    log: &Logger,
+) {
+    // elastic search url
+    const ES_URL: &str =
+        "https://elasticsearch.testnet.dfinity.network/ci-consensus-performance-test/_doc";
+    const NUM_UPLOAD_ATTEMPS: usize = 3;
+
+    let timestamp =
+        chrono::DateTime::<chrono::Utc>::from(std::time::SystemTime::now()).to_rfc3339();
+
+    let json_report = serde_json::json!(
+        {
+            "benchmark_name": "consensus_performance_test",
+            "timestamp": timestamp,
+            "ic_version": ic_version,
+            "benchmark_settings": {
+                "message_size": message_size,
+                "rps": rps,
+            },
+            "benchmark_results": {
+                "success_rate": metrics.success_rate,
+                "blocks_per_second": metrics.blocks_per_second,
+                "throughput_bytes_per_second": metrics.throughput_bytes_per_second,
+                "throughput_messages_per_second": metrics.throughput_messages_per_second,
+            }
+        }
+    );
+
+    info!(
+        log,
+        "Starting to upload performance test results to {ES_URL}: {}", json_report,
+    );
+
+    for i in 1..=NUM_UPLOAD_ATTEMPS {
```

Comment:
You could also consider using the [retry_with_msg!](https://github.com/dfinity/ic/blob/9ff9f96b038b38f56deda1ade528fe6ac3fa7794/rs/tests/driver/src/driver/test_env_api.rs#L2015) macro here.

Reply:
Good idea, done